moveit_kinematics/ikfast_kinematics_plugin/templates/ikfast61_moveit_plugin_template.cpp

```diff
@@ -544,12 +544,20 @@ int IKFastKinematicsPlugin::solve(KDL::Frame& pose_frame, const std::vector<doub
       return solutions.GetNumSolutions();
 
     case IKP_TranslationXAxisAngle4D:
+      double roll, pitch, yaw;
+      pose_frame.M.GetRPY(roll, pitch, yaw);
```

Comment:
According to the documentation snippet I've pasted above, and to https://github.com/rdiankov/openrave/issues/596#issuecomment-403673192, I think just using `GetRPY()` is wrong for `Translation*AxisAngle4D`. The specific angle you pass into `ComputeIk` (here: `&roll`) should be the angle between the manipulator direction and the X axis of the  manipulator base link’s coordinate system. Same for `TranslationYAxisAngle4D` and `TranslationZAxisAngle4D`.